gems/smithy-client/lib/smithy-client/block_io.rb

```diff
@@ -19,6 +19,7 @@ def initialize(headers = nil, &block)
       # @return [Integer]
       def write(chunk)
         @block.call(chunk, @headers)
+        chunk.bytesize
```

Comment:
I think adding this fixes an issue I ran into before in the read() method below. I don't remember what that was but it was related to return types.

Reply:
Yeah, this was returning the result of calling the block before which was incorrect.

Reply:
I think the read() code below can be simplified then.